### PR TITLE
[Znote 1.5] Flags by Raggaer reworked

### PR DESCRIPTION
**Pull request by Tandloze**
What I did was reworking the code in general since it was messy. Cleaned it up by using Znote inbuild functions. Also by a great idea of Znote himself of using the znote AAC structure instead of altering the tfs tables.

### CODE REVIEW

File: characterprofile.php [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
+ 
+if ($config['log_ip']) 
+{
 	znote_visitor_insert_detailed_data(4);
 }
-if (isset($_GET['name']) === true && empty($_GET['name']) === false) {
+
+if (isset($_GET['name']) === true && empty($_GET['name']) === false) 
+{
 	$name = $_GET['name'];
```
Comment (Znote) [inline characterprofile.php]: Avoid storing user input data raw in variables. 
```php
$name = getValue($_GET['name']);
```

File: characterprofile.php [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
+	{	
+		if ($config['TFSVersion'] == 'TFS_10') 
+		{
+			$profile_data = user_character_data($user_id, 'account_id', 'name', 'level', 'vocation', 'health', 'healthmax', 'experience', 'mana', 'manamax', 'sex', 'lastlogin');
 			$profile_data['online'] = user_is_online_10($user_id);
-			if ($config['Ach'] == true) {
+			
+			if ($config['Ach']) 
+			{
 				$achievementPoints = mysql_select_single("SELECT SUM(`value`) AS `sum` FROM `player_storage` WHERE `key` LIKE '30___' AND `player_id`='$user_id'");
```
Comment (Znote) [inline characterprofile.php]: ```player_id`='$user_id'````
Player id is an integer, so cast it to int (or make sure it is an int) before using it in a query.

**Comment by Znote** [comment]
Could you make an option in config.php to enable or disable flags on website?

**Comment by Tandloze** [comment]
Znote pull this master piece in :+1:  haha fixed the things you requested and cleaned up some if statements.

**Pull request merged**